org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/ProjectTemplate.xtend

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2017 itemis AG (http://www.itemis.de) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.wizard.template
+
+import java.lang.annotation.Retention
+import java.lang.annotation.Target
+import org.eclipse.xtend.lib.macro.Active
+
+/**
+ * Annotation to define a project templates that is selected by the user in the new project wizard and declared in a
+ * language specific ProjectTemplateProvider.
+ * 
+ * @author Arne Deutsch - Initial contribution and API
+ * @since 2.14
+ */
+@Target(TYPE)
+@Retention(RUNTIME)
+@Active(ProjectTemplateProcessor)
+annotation ProjectTemplate {
+	/**
+	 * Label of the project template presented to the user in the list of templates.
+	 */
+	String label
```

Comment:
How does this work with i18n?

Reply:
As the templates are java code I guess the labels can be extracted to property files the same as with any other code.